torchtitan/datasets/tokenizer/tiktoken.py

```diff
@@ -190,3 +195,30 @@ def _split_whitespaces_or_nonwhitespaces(
                     slice_start = i
                     current_slice_len = 1
         yield s[slice_start:]
+
+    def encode_multimodal(self, sample: Mapping[str, Any]) -> List[int]:
+        """
+        Tokenizes a `str` of text and creates `labels` masking BOS, EOS and `image_id` tokens.
+        """
+        # TODO(tj.solergibert) Should we keep `input_ids` OR `tokens` across this class, VisionCrossAttentionMask & the collator?
+        # For me it makes more sense to split `tokens` between `input_ids` & `labels` as in train.py BUT the `MultimodalDecoder`
+        # & everything else expects `tokens`
+        text = sample["text"]
+        tokens = self.encode(
+            text, bos=True, eos=True, allowed_special=set(["<|image|>"])
+        )
+        input_ids = tokens[:-1]
+        labels = tokens[1:]
```

Comment:
what if the first token is null? And for `HuggingFaceM4/OBELICS` specifically, "text" is a list of string or null. We don't need special treatment of it?

Reply:
We take care of the null/None values in the [`format_obelics`](https://github.com/TJ-Solergibert/torchtitan/blob/07a7a12af64075225874b18c99ab158776c2d254/torchtitan/datasets/multimodal/utils.py#L438) function. 

This function produces:
- `images`: List of decoded images in the sample
- `text`: str with the text of the sample ready to be tokenized **including the image tokens** (the null values you are referring)

So the `text` is ready to be tokenized!